pkg/password/password.go

```diff
@@ -0,0 +1,133 @@
+package password
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/store"
+	"github.com/replicatedhq/kots/pkg/util"
+	"golang.org/x/crypto/bcrypt"
+	corev1 "k8s.io/api/core/v1"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	ErrCurrentPasswordDoesNotMatch  = errors.New("current password does not match")
+	ErrNewPasswordTooShort          = errors.New("new password must be at least 6 characters")
+	ErrNewPasswordShouldBeDifferent = errors.New("new password should be different from current password")
+)
+
+// ValidatePasswordInput - will validate length and complexity of new password and check if it is different from current password
+func ValidatePasswordInput(currentPassword string, newPassword string) error {
+	if len(newPassword) < 6 {
+		return ErrNewPasswordTooShort
+	}
+
+	if newPassword == currentPassword {
+		return ErrNewPasswordShouldBeDifferent
+	}
+	return nil
+}
+
+// ValidateCurrentPassword - will compare the password with the stored password and return an error if they don't match
+func ValidateCurrentPassword(kotsStore store.Store, currentPassword string) error {
```

Comment:
needs a mutex

Reply:
I personally looked at this and thought the same thing at first, but this can be called from both the CLI and the server itself so IMO a mutex is unlikely to help prevent races

Can't hurt though

Reply:
lgtm although its possible to use resource version in k8s object as a mutex

somthing like validateAndSetSharedPassword

Reply:
lgtm as i dont think a mutex is really needed but maybe there is some way to use resource version mismatch as a mutex?

Reply:
resource reconciliation would be one good option, but that one is complicated.